### PR TITLE
Fix cursor routing for the MDV Lilli display driver

### LINKED ISSUE #7469: Braille Cursor routing bring to the next characther
### Steps to reproduce:
Open a text editor, I tried notepad, thunderbird new message and this edit field on Google Chrome.
Press a cursor routing.
The cursor will be positioned a charachter after your desired position.
For example if you have the word NVDA, and you want the cursor to the D charachter, after pressing cursor routing you will be positioned on the A charachter.
This behaviour is present also if you select braille tether review cursor, instead of focus.

### Expected behavior:
Cursor should go to the proper charachter.

### Actual behavior:
Cursor's position is one charachter after you expect.

### System configuration:
NVDA version: next-14283,1e5c5581 (August 2, 2017).

NVDA Installed or portable: portable.

Other information:
running windows7 64bit with a Lilli display braille.

Name and version of other software in use when reproducing the issue:

Windows Notepad, Thunderbird38.

### Other questions:

Does the issue still occur after restarting your PC? Yes.

Have you tried any other versions of NVDA? YES, 2017.2 is broken too.

### PR DESCRIPTION
**Pull request by LeonarddeR**
### Link to issue number:
Fixes #7469 

### Summary of the issue:
For MDV Lilli displays, using a cursor routing key moves the cursor one position ahead of the actual cell it should be positioned.

### Description of how this pull request fixes the issue:
The routing indexes started at 1, but they should be zero based. Subtracted an additional 1 in index calculation. Also, converted some decimal values to hexadecimal values to improve readability

### Testing performed:
@dreinn tested this, see https://github.com/nvaccess/nvda/issues/7469#issuecomment-319910224

### CODE REVIEW

File: source/brailleDisplayDrivers/lilli.py [hunk 1]
```diff
@@ -76,14 +76,14 @@ def _handleKeyPresses(self):
 			except:
 				pass
 			if not key: break
-			if (key <= 64) or ((key >= 257) and (key <= 296)): 
+			if (key <= 0x40) or ((key >= 0x101) and (key <= 0x128)):
```
Comment (dkager) [inline source/brailleDisplayDrivers/lilli.py]: You don't need parentheses around the comparisons.

Also, the second half could be written more compactly:
`0x101 <= key <= 0x128`
I find that a bit more readable. Also, this means you can remove all parentheses.

There is trailing whitespace here.

File: source/brailleDisplayDrivers/lilli.py [hunk 2]
```diff
@@ -76,14 +76,14 @@ def _handleKeyPresses(self):
...
-			if (key <= 64) or ((key >= 257) and (key <= 296)): 
+			if (key <= 0x40) or ((key >= 0x101) and (key <= 0x128)): 
 				self._onKeyPress(key)
 
 	def _onKeyPress(self, key):
 		try:
-			if (key >= 257) and (key <= 296):                                 
-				inputCore.manager.executeGesture(InputGesture(LILLI_KEYS[65],key-256))
-			elif (key <= 64):
+			if (key >= 0x101) and (key <= 0x128):
```
Comment (dkager) [inline source/brailleDisplayDrivers/lilli.py]: All three comments from above apply here as well.

File: source/brailleDisplayDrivers/lilli.py [hunk 3]
```diff
@@ -76,14 +76,14 @@ def _handleKeyPresses(self):
...
+			if (key <= 0x40) or ((key >= 0x101) and (key <= 0x128)): 
 				self._onKeyPress(key)
 
 	def _onKeyPress(self, key):
 		try:
-			if (key >= 257) and (key <= 296):                                 
-				inputCore.manager.executeGesture(InputGesture(LILLI_KEYS[65],key-256))
-			elif (key <= 64):
+			if (key >= 0x101) and (key <= 0x128):                                 
+				inputCore.manager.executeGesture(InputGesture(LILLI_KEYS[65],key-0x101))
```
Comment (dkager) [inline source/brailleDisplayDrivers/lilli.py]: I would replace 65 by a constant, e.g. `LILLI_ROUTING`.

File: source/brailleDisplayDrivers/lilli.py [hunk 4]
```diff
@@ -76,14 +76,14 @@ def _handleKeyPresses(self):
...
+			elif (key <= 0x40):
```
Comment (dkager) [inline source/brailleDisplayDrivers/lilli.py]: Parentheses. Yes, being very picky. :)

**Review by dkager — CHANGES_REQUESTED** [review]

**Comment by leonardder** [comment]
@dkager: Are you willing to do these changes?

**Review by dkager — APPROVED** [review]

**Review by michaelDCurran — APPROVED** [review]

**Pull request merged**